### PR TITLE
chore(ME-3491): add private network IP to socket config

### PR DESCRIPTION
**Pull request by btoonk**
## Summary by CodeRabbit

- **New Features**
	- Introduced support for specifying and validating private network IPv4 and IPv6 addresses in the connector service configuration.
  
- **Bug Fixes**
	- Enhanced validation logic to ensure correct IP address formats for both IPv4 and IPv6 fields.

### CODE REVIEW

File: types/service/configuration.go [hunk 1]
```diff
@@ -91,12 +92,26 @@ type ConnectorServiceConfiguration struct {
...
+		if net.ParseIP(*c.PrivateNetworkIPv4) == nil {
+			return fmt.Errorf("invalid private network IPv4")
+		}
+	}
+
+	if c.PrivateNetworkIPv6 != nil {
+		if net.ParseIP(*c.PrivateNetworkIPv6) == nil {
+			return fmt.Errorf("invalid private network IPv6")
+		}
+	}
```
[the remark this answers is not in the document]
Comment (btoonk) [reply types/service/configuration.go]: changed

**Review by th3wingman — APPROVED** [review]
:ship:

File: types/service/configuration.go [hunk 2]
```diff
@@ -91,12 +92,50 @@ type ConnectorServiceConfiguration struct {
...
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateIP(ipStr *string, version string) error {
+	if ipStr == nil {
+		return nil
```
Comment (adrianosela) [inline types/service/configuration.go:122]: shouldnt this be an error if you expect non nil?

File: types/service/configuration.go [hunk 3]
```diff
@@ -91,12 +92,50 @@ type ConnectorServiceConfiguration struct {
 	EndToEndEncryptionEnabled      bool          `json:"end_to_end_encryption_enabled"`
 	RecordingEnabled               bool          `json:"recording_enabled"`
 	Upstream                       Configuration `json:"upstream_configuration"`
+	PrivateNetworkIPv4             *string       `json:"private_network_ipv4"`
```
Comment (adrianosela) [inline types/service/configuration.go:95]: I think using strings instea of ptr would be easier to manage, You can also add omitempty tags here.

**Pull request merged**